torch_xla/csrc/tensor.cpp

```diff
@@ -2257,10 +2260,10 @@ void XLATensor::ApplyPendingGraph(std::vector<XLATensor>* tensors,
       contexts_map.size());
   size_t index = 0;
   for (auto& device_and_context : contexts_map) {
-    const Device& device = device_and_context.first;
+    Device device = device_and_context.first;
     DeviceContext* device_context = &device_and_context.second;
 
-    auto generator = [&, device_context, index]() {
+    auto generator = [&, device, device_context, index]() {
```

Comment:
This mix of capture all and then only some is confusing. In such cases, I'd rather list what we need to capture.

Reply:
How listing 15 of them makes it better?